crates/language_models/src/provider/copilot_chat.rs

```diff
@@ -475,9 +458,43 @@ impl CopilotChatLanguageModel {
                         }
                     }
 
+                    let mut content_parts = Vec::new();
+                    let mut text_content = String::new();
+
+                    for content in &message.content {
+                        match content {
+                            MessageContent::Text(text) | MessageContent::Thinking { text, .. } => {
+                                if !text.is_empty() {
+                                    text_content.push_str(text);
+                                }
+                            }
+                            MessageContent::Image(image) => {
```

Comment:
```suggestion
                            MessageContent::Image(image) if self.model.supports_vision() => {
```
You can coalesce the if statement in the match expression, it makes it clearer that this case is not used otherwise.